### DISCUSSION TITLE
How to depend on only one specific key from input with MultiPartitionsDefinition?

### DISCUSSION THREAD

**Discussion by rprechelt** [dagster-io/dagster#18458] [Q&A] [answered]
I have a `MultiPartitionsDefinition` that is used by an `upstream` asset:

``` python
date_and_country = MultiPartitionsDefinition(
    {
        "date": DailyPartitionsDefinition(start_date="2022-03-12"),
        "country": StaticPartitionsDefinition(["USA", "Canada", "Australia"])
    })

@asset(partitions_def=date_and_country)
def upstream():
    pass
```

I have another `MultiPartitionsDefinition` that is used by a `downstream` asset:

``` python
date_and_color = MultiPartitionsDefinition(
    {
        "date": DailyPartitionsDefinition(start_date="2022-03-12"),
        "color": StaticPartitionsDefinition(["red", "green", "blue"])
    })

@asset(partitions_def=date_and_color)
def downstream(upstream):
    pass

```

I want `downstream` to receive the `upstream` asset for only a single "specific" key from the "country" dimension in `upstream`, i.e. `downstream` depends (and receives as input) only the "USA" slice of the `upstream` asset (while preserving the one-to-one mapping of the "date" dimension). I can't quite get this to work with the `MultiPartitionMapping`'s that I've tried. 

How can I achieve this?

**Comment by clairelin135** [comment, marked as answer]
@rprechelt In your case, you can provide a `SpecificPartitionsPartitionMapping` to `MultiPartitionMapping` that defines that each color partition depends on a specific partition in the upstream "country" dimension:

```python
@asset(
    partitions_def=date_and_color,
    ins={
        "upstream": AssetIn(
            partition_mapping=MultiPartitionMapping(
                {
                    "date": DimensionPartitionMapping(
                        "date", TimeWindowPartitionMapping()
                    ),
                    "country": DimensionPartitionMapping(
                        "color", SpecificPartitionsPartitionMapping(["USA"])
                    ),
                }
            ),
        )
    },
)
def downstream(context, upstream):
    context.log.info(context.asset_partition_keys_for_input("upstream"))
    ...

```

**Reply by rprechelt** [reply]
Great! This worked exactly as described.
